doc/west/repo-tool.rst

```diff
@@ -0,0 +1,270 @@
+<https://github.com/zephyrproject-rtos/west/blob/master/src/west/manifest-schema.yml>`_.
+
+The ``manifest-rev`` branch
+***************************
+
+West creates a branch named ``manifest-rev`` in each project, pointing to the
+commit at the project's revision, as specified in the manifest. The
+``manifest-rev`` branch is updated whenever project data is fetched (the
+`command overview`_ below explains which commands fetch project data).
+
+All branches created through West track the ``manifest-rev`` branch. Several
+multi-repo commands also use ``manifest-rev`` as a reference for the upstream
+revision (as of the most recent fetch).
+
+.. note::
+
+   ``manifest-rev`` is a normal Git branch, and is only treated specially by
+   name. If you delete it or otherwise modify it, it will be
+   recreated/reset when upstream data is next fetched by ``west``, as if
+   through ``git reset --hard`` (though ``git update-ref`` is used internally).
+
+   Since ``manifest-rev`` represents the upstream revision as of the most
+   recent fetch, it is normally a bad idea to modify it.
+
+   ``manifest-rev`` was added to allow branches to track SHA revisions, and to
+   give a consistent reference for the upstream revision regardless of how the
+   manifest changes over time.
+
+Command overview
+================
+
+This section gives a quick overview of the multi-repo commands, split up by
+functionality. All commands loosely mimic the corresponding Git command, but in
+a multi-repo context (``west fetch`` fetches upstream data without updating
+local work branches, etc.)
+
+Passing no projects to commands that accept a list of projects usually means to
+run the command for all projects listed in the manifest.
+
+.. note::
+
+   For the most up-to-date documentation, see the command help texts (e.g.
+   ``west fetch --help``). Only the most important flags are mentioned here.
+
+Cloning and updating projects
+*****************************
+
+After running ``west init`` to initialize West (e.g., with the default Zephyr
+manifest), the following commands will clone/update projects.
+
+Except for ``west rebase``, all commands in this section implicitly update the
+manifest repository, fetch upstream data, and update the ``manifest-rev``
+branch.
+
+- ``west clone [-b BRANCH_NAME] [PROJECT ...]``: Clones the specified
+  projects (default: all projects).
+
+  An initial branch named after the project's manifest revision is created in
+  each cloned repository. The names of branch and tag revisions are used as-is.
+  For qualified refs like ``refs/heads/foo``, the last component (``foo``) is
+  used. For SHA revisions, the generic name ``work`` is used.
+
+  A different branch name can be specified with ``-b``.
+
+  Like all branches created through West, the initial branch tracks
+  ``manifest-rev``.
+
+- ``west fetch [PROJECT ...]``: Fetches new upstream changes in each of the
+  specified projects (default: all projects), cloning them first if necessary.
```

Comment:
Maybe mention that this is like `git fetch`, i.e. does not modify local branches, just `<remote>/*` branches.

Reply:
Now mentions that local branches aren't updated (except for `manifest-rev`).